src/v/cloud_storage/tests/cloud_storage_e2e_test.cc

```diff
@@ -672,3 +673,164 @@ FIXTURE_TEST(
+
+    // Force local timequeries only through archival mode.
+    cluster::topic_properties props;
+    props.shadow_indexing = model::shadow_indexing_mode::archival;
+
+    add_topic({model::kafka_namespace, topic_name}, 1, props).get();
+
+    wait_for_leader(ntp).get();
+
+    auto partition = app.partition_manager.local().get(ntp);
+    auto log = partition->log();
+    auto& archiver = partition->archiver().value().get();
+    BOOST_REQUIRE(archiver.sync_for_tests().get());
+
+    const auto batches_per_segment = 1;
+    const auto num_segs = 5;
+    const auto batch_time_delta_ms = 10;
+    const auto base_timestamp = model::timestamp{0};
+    tests::remote_segment_generator gen(make_kafka_client().get(), *partition);
+    auto total_records = gen.num_segments(num_segs)
+                           .batches_per_segment(batches_per_segment)
+                           .base_timestamp(base_timestamp)
+                           .batch_time_delta_ms(batch_time_delta_ms)
+                           .produce()
+                           .get();
+    BOOST_REQUIRE_EQUAL(total_records, 5);
+
+    auto make_and_verify_timequery =
+      [partition](
+        model::timestamp t,
+        model::offset o,
+        bool expect_value = false,
+        std::optional<model::offset> expected_o = std::nullopt) {
+          auto timequery_conf = storage::timequery_config(
+            t, o, ss::default_priority_class(), std::nullopt);
+
+          auto result = partition->timequery(timequery_conf).get();
+
+          if (expect_value) {
+              BOOST_REQUIRE(result.has_value());
+              BOOST_REQUIRE_EQUAL(result.value().offset, expected_o.value());
+          } else {
+              BOOST_REQUIRE(!result.has_value());
+          }
+      };
+
+    make_and_verify_timequery(
+      base_timestamp, model::offset{0}, true, model::offset{0});
+
+    for (int i = 1; i < total_records; ++i) {
+        const auto min_timestamp = base_timestamp()
+                                   + batch_time_delta_ms * (i - 1);
+        const auto max_timestamp = min_timestamp + batch_time_delta_ms;
+        const auto query_timestamp = random_generators::get_int(
+          min_timestamp + 1, max_timestamp);
+        make_and_verify_timequery(
+          model::timestamp{query_timestamp},
+          model::offset{i},
+          true,
+          model::offset{i});
+    }
+
+    make_and_verify_timequery(
+      model::timestamp{
+        base_timestamp() + (batch_time_delta_ms * total_records)},
+      model::offset{total_records},
+      false);
+}
+
+FIXTURE_TEST(test_cloud_storage_timequery, e2e_fixture) {
+    const model::topic topic_name("tapioca");
+    model::ntp ntp(model::kafka_namespace, topic_name, model::partition_id{0});
+
+    // Force local timequeries only through archival mode.
+    cluster::topic_properties props;
+    props.shadow_indexing = model::shadow_indexing_mode::full;
```

Comment:
Meant to be archival mode?

Reply:
Oops, line 753 should not have been copy pasted here.